internal/brain/suggestor.go

```diff
@@ -1,40 +1,110 @@
 package brain
 
 import (
-	"math/rand"
+	"fmt"
+	"time"
 
 	"github.com/gogogomoku/gomoku/internal/board"
+	tr "github.com/gogogomoku/gomoku/internal/tree"
 )
 
-func SuggestMove() {
-	possible := []int{}
-	best := []int{}
+type Move struct {
+	Position int
+	Value    int
+}
+
+func getPossibleMoves(node *tr.Node) []Move {
+	poss := []Move{}
 	for i := 0; i < (board.SIZE * board.SIZE); i++ {
-		if GameRound.Goban.Tab[i] == 0 {
-			if CheckValidMove(i) {
-				seq := CompleteSequenceForPosition(i, GameRound.CurrentPlayer.Id)
-				if len(seq) > 0 {
-					best = append(best, i)
+		if node.Tab[i] == 0 {
+			if CheckValidMove(i, node.Tab) {
+				// Add move only if it is aligned in a range of 5 from other
+				affectsTab := false
+				lines := CheckNextN(i, node.Tab, 1)
+				for _, line := range lines {
+					for _, piece := range line {
+						if piece != 0 {
+							affectsTab = true
+							break
+						}
+					}
+				}
+				if affectsTab {
+					poss = append(poss, Move{Position: i, Value: 0})
 				}
-				possible = append(possible, i)
 			}
 		}
 	}
-	ran := 0
-	if len(best) > 0 {
-		if len(best)-1 < 2 {
-			ran = 0
-		} else {
-			ran = int(rand.Intn(len(best) - 1))
+	return poss
+}
+
+func addNewLayer(poss []Move, node *tr.Node, playerId int) {
```

Comment:
This function just stores all legal and "possible" (part of a potential 5-len sequence) single-step moves for a given player (please correct me/comment code if I am wrong)?